helpers/compile/plugins/depCheckPlugin.ts

```diff
@@ -94,25 +90,13 @@ const unusedPlugin: esbuild.Plugin = {
 
       // we exclude the deps that match our unusedIgnore patterns
       const filteredMissingDeps = missingDependencies.filter((dep) => {
-        return !missingIgnore.some((pattern) => dep.match(pattern))
+        return !missingIgnore.some((pattern) => dep.match(pattern)) && !peerDependencies.includes(dep)
       })
 
-      console.warn('unusedDependencies', filteredUnusedDeps)
-      console.warn('missingDependencies', filteredMissingDeps)
+      console.warn('unusedDependencies', JSON.stringify(filteredUnusedDeps))
+      console.warn('missingDependencies', JSON.stringify(filteredMissingDeps))
+
+      if (filteredMissingDeps.length > 0) process.exit(1)
     })
   },
-}
-
-void esbuild
-  .build({
-    entryPoints: glob.sync('**/*.{j,t}s', {
-      ignore: ['**/packages/**/*', '**/*.d.ts'],
-      gitignore: true,
-    }),
-    logLevel: 'silent', // there will be errors
-    bundle: true, // we bundle to get everything
-    write: false, // no need to write for analysis
-    outdir: 'out',
-    plugins: [unusedPlugin],
-  })
-  .catch(() => {})
```

Comment:
Where does this logic go?

Reply:
It moved to `prisma/helpers/compile/build.ts`